Document the repo/workspace (#35)
getsentry/ophio: README.md

See also [RFC119](https://github.com/getsentry/rfcs/pull/119) for more details.

## How to write Bindings

> - If you want to expose existing functionality within a `crates.io` crate:

Reviewer: This sentence initially confused me, I think it's clearer this way:
```suggestion
- If you want to expose functionality from an existing crate:
```

- Otherwise, if this is purpose built code not intended for usage outside of Sentry:
  - Write (and test) straight Rust code in the `rust/` crate.
  - Write bindings for it in `bindings/`.
- Re-export all the code from `bindings/` within `python/sentry_ophio`.
- If necessary, write custom `pyi` typings in `python/sentry_ophio`.
- Write some `pytest` tests the `tests/`.
- After publishing, you import the code in the main `sentry` repo from the `sentry_ophio` package.

For more details, consult the [Repo Structure](#repo-structure) section.

## Publishing

The release / publish workflow is fully set up in the repo.

We do not maintain any kind of `semver` guarantees whatsoever.
For this reason, one can just increment the minor version for every new release / publish.

- Manually trigger the [`Release workflow`](actions/workflows/release.yml)
- This should ideally auto-approve the publish, publish to `pypi` and then uplift that to our
  internal `pypi` mirror as well.
- If the publish fails, which it unfortunately does way too often, search for a soundproof booth
  and scream from the top of your lungs, then go to `#z-vent` and complain loudly about

>   the broken publishes and that we can’t have nice things.

Author: ![image](https://github.com/getsentry/ophio/assets/580492/0d596f28-6c55-4464-9c1b-7b00136e4c89)

Reviewer: ![77099e99060209eda33a58858fd7b962](https://github.com/getsentry/ophio/assets/7622248/d0829901-0d75-4bf6-a4a6-2303d10ae773)
